core/score_sync.go

```diff
@@ -0,0 +1,82 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"application/core/model"
+	"application/driven/storage"
+	"log"
+)
+
+// SyncAmgUUIDForScore fetches the AmgUUID from Core BB and populates the external_user_id field
+func (app *Application) SyncAmgUUIDForScore(score *model.Score, externalIDs map[string]string) error {
+	// Get Mastodon ID from externalIDs if provided, otherwise use the one already in score
+	mastodonID := score.ExternalProfileID
+	if externalIDs != nil {
+		if mastodonIDFromMap, ok := externalIDs["mastodon_id"]; ok && mastodonIDFromMap != "" {
+			mastodonID = mastodonIDFromMap
+		}
+	}
+
+	if mastodonID == "" {
+		log.Printf("SyncAmgUUIDForScore: no mastodon_id provided for score, skipping AmgUUID sync")
+		return nil
+	}
+
+	// Look up the AmgUUID from Core BB
+	accountCriteria := storage.BuildCoreAccountCriteriaByMastodonID(mastodonID)
+
+	coreAccounts, err := app.corebb.RetrieveCoreUserAccountByCriteria(accountCriteria, &score.AppID, &score.OrgID)
+	if err != nil {
+		log.Printf("SyncAmgUUIDForScore: error retrieving Core account: %v", err)
+		// Don't fail the score creation if Core BB lookup fails, just log it
+		return nil
+	}
+
+	if len(coreAccounts) == 0 {
+		log.Printf("SyncAmgUUIDForScore: no Core account found for mastodon_id %s", mastodonID)
+		return nil
+	}
+
+	if len(coreAccounts) > 1 {
+		log.Printf("SyncAmgUUIDForScore: WARNING - multiple Core accounts found for mastodon_id %s, using first one", mastodonID)
+	}
+
+	// Set the AmgUUID
+	score.ExternalUserID = coreAccounts[0].ID
+	log.Printf("SyncAmgUUIDForScore: synced AmgUUID %s for mastodon_id %s", score.ExternalUserID, mastodonID)
+
+	return nil
+}
+
+// PrepareScoreForResponse modifies a score for API response based on feature flag
+func (app *Application) PrepareScoreForResponse(score *model.Score) {
+	if app.shouldUseExternalUserID() && score.ExternalUserID != "" {
+		// When flag is enabled and we have an AmgUUID, use it in the external_profile_id field
+		score.ExternalProfileID = score.ExternalUserID
+	}
+}
+
+// PrepareScoresForResponse prepares multiple scores for API response
+func (app *Application) PrepareScoresForResponse(scores []model.Score) {
+	for i := range scores {
+		app.PrepareScoreForResponse(&scores[i])
+	}
+}
+
+// shouldUseExternalUserID returns the feature flag value
+func (app *Application) shouldUseExternalUserID() bool {
+	return app.useExternalUserID
+}
```

Comment:
This function is unnecessary because it is only used in the `application` package, so you can directly access `app.useExternalUserID`.